brave/browser-laptop
# Fixes wallet recovery dialogue re-appearing on relaunch after a successful recovery.

Fixes #13645

Additionally, per the feedback on this [PR](https://github.com/brave/browser-laptop/pull/13647):

1. about:preferences#payments is maintained when the recovery dialogue is closed

## Submitter Checklist:


## Test Plan:

1. Clean install
2. Create Wallet in about:preferences#payments
3. Exit Brave
4. Delete seed 20 and 21 in ledger-state.json
5. Run Brave again
6. Click "Recover your Brave wallet"
7. Dismiss the modal dialogue via the 'x' or cancel button.
8. Ensure that about:preferences#payments is opened
9. Then successfully recover the wallet with:
`vower obovoid menace tobogganist hoyle honoree pixel pestilently disconcertment sellable ruffing supervision zoroastrian based coparent slackened`
10. Click OK on the dialog
11. Relaunch Brave
12. Ensure that about:preferences#payments is opened
13. The same expected behavior should occur when using the Brave Wallet Backup dialog
14. Open the back up dialog, then press 'x' to close it out
15. Ensure that about:preferences#payments is opened
16. Go back to the backup dialogue, this time hitting 'Done'
17. Ensure that about:preferences#payments is opened

**Links:** 
recovery: `about:preferences#payments?ledgerRecoveryOverlayVisible`
backup: `about:preferences#payments?ledgerBackupOverlayVisible`

## Reviewer Checklist:


Tests


cc: @bsclifton @btlechowski
